server/src/main/java/com/google/fhir/gateway/AllowedQueriesChecker.java

```diff
@@ -118,4 +125,15 @@ private boolean requestMatches(RequestDetailsReader requestDetails, AllowedQuery
         "Allowed-queries entry {} matched query {}", entry, requestDetails.getCompleteUrl());
     return true;
   }
+
+  private boolean allowRequestPath(String path, AllowedQueryEntry entry) {
+    if (path.equals(entry.getPath())) {
+      return true;
+    }
+    if (!AllowedQueriesConfig.MATCHES_PATH_ANY_VALUE_PATTERN.matcher(entry.getPath()).matches()) {
+      return false;
+    }
+    String basePath = entry.getPath().substring(0, entry.getPath().indexOf("/") + 1);
+    return path.startsWith(basePath);
```

Comment:
For the new test config you have added, does this allow `Composition/` (i.e., without anything else after `/`)? If yes, why not allow `Composition` too?

Reply:
right. I will fix this after concluding the above conversation.

Reply:
done